Cannot disable Spinel debug logging

I have built using 
```OTBR_OPTIONS=-DOT_LOG_LEVEL=INFO ...```

However my resultant output still contains a ton of 
```
otbr                 | Jun  7 13:18:18 c66b5ad37c98 otbr-agent[158]: 00:00:44.796 [D] P-RadioSpinel-: Sent spinel frame, flg:0x2, iid:0, tid:13, cmd:PROP_VALUE_SET, key:PHY_CHAN, channel:11
otbr                 | Jun  7 13:18:18 c66b5ad37c98 otbr-agent[158]: 00:00:44.796 [D] P-RadioSpinel-: Wait response: tid=13 key=33

```
Is there a bug in the cmake stuff? I don't know how to fix it but it seems it does not pass the OT_LOG_LEVEL into the third_party openthread build which is where the radio_spinel.cpp code is.

Hi @Irving-cl,
I updated to the latest
```
ot-ctl 
version
OPENTHREAD/32f462f; POSIX; Jun 13 2024 12:32:59
```
Same issue with several debug level items:
```
otbr  | Jun 13 12:40:32 7200f296ca22 otbr-agent[163]: 00:00:43.910 [D] Cli-----------: Output: Done
...
otbr  | Jun 13 12:40:48 7200f296ca22 otbr-agent[163]: 00:01:00.015 [D] P-SpinelDrive-: Received spinel frame, flg:0x2, iid:0, tid:10, cmd:PROP_VALUE_IS, key:TIMESTAMP, timestamp:4068868340
...
otbr  | Jun 13 12:41:17 7200f296ca22 otbr-agent[163]: 00:01:28.917 [D] P-RadioSpinel-: Wait response: tid=7 key=33
...
```
I don't think the static_assert did what we want...
```
#9 140.6 /app/third_party/openthread/repo/src/core/instance/instance.cpp:73:25: error: 'ot::LogLevel ot::Instance::sLogLevel' is private within this context
#9 140.6    73 | static_assert(Instance::sLogLevel == OT_LOG_LEVEL_INFO, "Initial log level incorrect");
#9 140.6       |                         ^~~~~~~~~
#9 140.6 compilation terminated due to -Wfatal-errors.

```

```
#ifdef OPENTHREAD_CONFIG_LOG_LEVEL
#if OPENTHREAD_CONFIG_LOG_LEVEL == OT_LOG_LEVEL_INFO
#warning "LOG_LEVEL is INFO"
#elif OPENTHREAD_CONFIG_LOG_LEVEL == OT_LOG_LEVEL_DEBG
#warning "LOG_LEVEL is DEBG"
#elif OPENTHREAD_CONFIG_LOG_LEVEL == OT_LOG_LEVEL_CRIT
#warning "LOG_LEVEL is CRIT"
#elif OPENTHREAD_CONFIG_LOG_LEVEL == OT_LOG_LEVEL_WARN
#warning "LOG_LEVEL is WARN"
#elif OPENTHREAD_CONFIG_LOG_LEVEL == OT_LOG_LEVEL_NOTE
#warning "LOG_LEVEL is NOTE"
#elif OPENTHREAD_CONFIG_LOG_LEVEL == OT_LOG_LEVEL_NONE
#warning "LOG_LEVEL is NONE"
#else
#warning "LOG_LEVEL is something else"
#endif
#else
#warning "OPENTHREAD_CONFIG_LOG_LEVEL not defined"
#endif

```
Results in
```
#9 54.10 In file included from /app/third_party/openthread/repo/src/core/openthread-core-config.h:68,
#9 54.10                  from /app/third_party/openthread/repo/src/posix/platform/openthread-posix-config.h:32,
#9 54.10                  from /app/third_party/openthread/repo/src/posix/platform/platform-posix.h:38,
#9 54.10                  from /app/third_party/openthread/repo/src/posix/platform/radio_url.cpp:36:
#9 54.10 /app/third_party/openthread/repo/src/posix/platform/openthread-core-posix-config.h:57:2: error: #warning "LOG_LEVEL is INFO" [-Werror=cpp]
#9 54.10    57 | #warning "LOG_LEVEL is INFO"
#9 54.10       |  ^~~~~~~
#9 54.10 compilation terminated due to -Wfatal-errors.

```

And even though I pass -DOT_COMPILE_WARNING_AS_ERROR=OFF as --build-arg, it still does not disable it. See about -
Wfatal-errors.
```
#9 53.04 ++ OTBR_OPTIONS='-DOT_COMPILE_WARNING_AS_ERROR=OFF -DOT_LOG_LEVEL=INFO -DOT_CHANNEL_MONITOR=ON -DOT_CHANNEL_MANAGER=ON -DOT_LINK_METRICS_MANAGER=ON -DOT_LINK_METRICS_SUBJECT=ON'
```

No I did not change some source code, I was just trying to use multiple #ifdef #warnings but the first one stops compilation.

--- a/etc/docker/docker_entrypoint.sh
+++ b/etc/docker/docker_entrypoint.sh
@@ -58,6 +58,11 @@ function parse_args()
                 shift
                 shift
                 ;;
+            --debug-level)
+                DEBUG_LEVEL=$2
+                shift
+                shift
+                ;;
             *)
                 shift
                 ;;
@@ -81,12 +86,14 @@ parse_args "$@"
 [ -n "$TUN_INTERFACE_NAME" ] || TUN_INTERFACE_NAME="wpan0"
 [ -n "$BACKBONE_INTERFACE" ] || BACKBONE_INTERFACE="eth0"
 [ -n "$NAT64_PREFIX" ] || NAT64_PREFIX="64:ff9b::/96"
+[ -n "$DEBUG_LEVEL" ] || DEBUG_LEVEL="7"
 
 echo "RADIO_URL:" $RADIO_URL
 echo "TREL_URL:" "$TREL_URL"
 echo "TUN_INTERFACE_NAME:" $TUN_INTERFACE_NAME
 echo "BACKBONE_INTERFACE: $BACKBONE_INTERFACE"
 echo "NAT64_PREFIX:" $NAT64_PREFIX
+echo "DEBUG_LEVEL:" $DEBUG_LEVEL
 
 NAT64_PREFIX=${NAT64_PREFIX/\//\\\/}
 TAYGA_CONF=/etc/tayga.conf
@@ -96,8 +103,8 @@ BIND_CONF_OPTIONS=/etc/bind/named.conf.options
 ! test -f $BIND_CONF_OPTIONS || sed -i "s/dns64.*$/dns64 $NAT64_PREFIX {};/" $BIND_CONF_OPTIONS
 sed -i "s/$INFRA_IF_NAME/$BACKBONE_INTERFACE/" /etc/sysctl.d/60-otbr-accept-ra.conf
 
-echo "OTBR_AGENT_OPTS=\"-I $TUN_INTERFACE_NAME -B $BACKBONE_INTERFACE -d7 $RADIO_URL $TREL_URL\"" >/etc/default/otbr-agent
-echo "OTBR_WEB_OPTS=\"-I $TUN_INTERFACE_NAME -d7 -p 80\"" >/etc/default/otbr-web
+echo "OTBR_AGENT_OPTS=\"-I $TUN_INTERFACE_NAME -B $BACKBONE_INTERFACE -d${DEBUG_LEVEL} $RADIO_URL $TREL_URL\"" >/etc/default/otbr-agent
+echo "OTBR_WEB_OPTS=\"-I $TUN_INTERFACE_NAME -d${DEBUG_LEVEL} -p 80\"" >/etc/default/otbr-web
 
 /app/script/server
 

